test/coreneuron/test_spikes.py

```diff
@@ -1,8 +1,16 @@
+import os
 import pytest
-import sys
 
+# Hacky, but it's non-trivial to pass commandline arguments to pytest tests.
```

Comment:
Just curious.. What was the issue with the arguments? IIUC they were working before(?)

Reply:
I assume Olli is referring to executions when you launch pytest and then it discovers & runs all tests. (i.e. without calling `python file <args>`)

Reply:
The issue is that
```
python -m pytest [pytest options] path/to/test.py --arg
```
does not forward `--arg`to `path/to/test.py`, it tries to open a file called `--arg`. [You can work around this by telling `pytest` about the options you are going to use](https://github.com/neuronsimulator/nrn/blob/eaee9620c1804cab9e15af455e376849fd090c1d/test/rxd/conftest.py#L9-L11), but I was defining this as "non-trivial". Particularly as (see below) we also have to launch these tests without `pytest`.

For the tests I am actually touching here, before we had:
- [`python -m pytest --cov-report=xml --cov=neuron test/coreneuron/test_spikes.py`](https://github.com/neuronsimulator/nrn/blob/91b246c2ee04d5033fa359bd1a4ba96d64c8dfc7/test/CMakeLists.txt#L202)
- [`python test/coreneuron/test_spikes.py file_mode`](https://github.com/neuronsimulator/nrn/blob/91b246c2ee04d5033fa359bd1a4ba96d64c8dfc7/test/CMakeLists.txt#L208)
- [`mpiexec -n 2 python test/coreneuron/test_spikes.py mpi4py`](https://github.com/neuronsimulator/nrn/blob/91b246c2ee04d5033fa359bd1a4ba96d64c8dfc7/test/CMakeLists.txt#L255-L256)
- [`mpiexec -n 2 special -python -pyexe path/to/python test/coreneuron/test_spikes.py nrnmpi_init file_mode`](https://github.com/neuronsimulator/nrn/blob/91b246c2ee04d5033fa359bd1a4ba96d64c8dfc7/test/CMakeLists.txt#L264-L265)
i.e. we had `-m pytest` or commandline arguments, but not both.

With this PR then we flip between `special -python` and `python` depending on whether GPU support is enabled, and it made it a little simpler to parametrise that (`${modtests_launch_{py,hoc,py_mpi}}`) if the tests were launched uniformly. So now 1-3 of the above list have `-m pytest` instead of just #1.